Makefile

```diff
@@ -37,6 +37,27 @@ ct-tls:
 edoc:
 	@rebar3 edoc skip_deps=true
 
+publish: edoc patch-doc
+	mix docs
+	mix hex.publish package
+
+# Patch the markdown doc generated by edown. This is needed to be able
+# to generate documentation via Elixirs mix.
+# Following changes are needed:
+# - Handle links in headers, change '### <a name="link">Header</a> ###' to
+#                                   '<a name="link"></a> ### Header ###'
+# - Newline needed for before following tags:
+#   </table> </dd> </pre>
+# - Removal of unneeded line breaks (visual only)
+patch-doc: doc/*.md
+	@for file in $^ ; do \
+		sed -i 's|### <a name="\(.*\)">\(.*\)</a> ###|<a name="\1"></a>\n### \2 ###|g' $${file} ; \
+		sed -i 's|</table>|\n</table>|g' $${file} ; \
+		sed -i 's|</dd>|\n</dd>|g' $${file} ; \
+		sed -i 's|</code></pre>|</code>\n</pre>|g' $${file} ; \
+		sed -i 's|<br />||g' $${file} ; \
+	done
+
```

Comment:
I had to change the `sed` command to work on OSX: 

```diff
diff --git a/Makefile b/Makefile
index c779777..7d5ffeb 100644
--- a/Makefile
+++ b/Makefile
@@ -51,11 +51,11 @@ publish: edoc patch-doc
 # - Removal of unneeded line breaks (visual only)
 patch-doc: doc/*.md
        @for file in $^ ; do \
-               sed -i 's|### <a name="\(.*\)">\(.*\)</a> ###|<a name="\1"></a>\n### \2 ###|g' $${file} ; \
-               sed -i 's|</table>|\n</table>|g' $${file} ; \
-               sed -i 's|</dd>|\n</dd>|g' $${file} ; \
-               sed -i 's|</code></pre>|</code>\n</pre>|g' $${file} ; \
-               sed -i 's|<br />||g' $${file} ; \
+               sed -i '' 's|### <a name="\(.*\)">\(.*\)</a> ###|<a name="\1"></a>\n### \2 ###|g' $${file} ; \
+               sed -i '' 's|</table>|\n</table>|g' $${file} ; \
+               sed -i '' 's|</dd>|\n</dd>|g' $${file} ; \
+               sed -i '' 's|</code></pre>|</code>\n</pre>|g' $${file} ; \
+               sed -i '' 's|<br />||g' $${file} ; \
        done

 xref:
```

source: https://myshittycode.com/2014/07/24/os-x-sed-extra-characters-at-the-end-of-l-command-error/

Reply:
If it works, I'm OK with adding it. I think it needs a code comment to explain what the `''` is though, or add a backup suffix to `-i` (e.g. `-i.backup`). It seems like the optional backup suffix which follows `-i` is mandatory in Mac. See also this: https://stackoverflow.com/questions/7573368/in-place-edits-with-sed-on-os-x#7573438

(Sad story: Mac OSX is stuck with old versions of sed, because recent versions of sed are GPL 3 and Apple refuse to include GPL 3 stuff in their OS. It's the same with all GNU utils.)

Reply:
Had to use the -i.bak variant due to that -i '' did not work on Linux. I tested the -i.bak variant on mac and that seemed to work there aswell